### PR TITLE
Feature/boosttransaction

### PR DESCRIPTION
**Pull request by andyoknen**

### CODE REVIEW

File: src/pocketdb/web/SearchRpc.cpp [hunk 1]
```diff
@@ -285,6 +285,8 @@ namespace PocketWeb::PocketWebRpc
             throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, string("Invalid Pocketcoin address: ") + address);
 
         int cntOut = 10;
+        if (request.params.size() > 1 && request.params[1].isNum())
```
Comment (tawmaz) [inline src/pocketdb/web/SearchRpc.cpp:288]: Seems like we should check require.params.size() at line 279 to make sure we have arguments and determine which parameters to process.

File: src/pocketdb/web/SearchRpc.cpp [hunk 2]
```diff
@@ -334,6 +345,15 @@ namespace PocketWeb::PocketWebRpc
         int depth = 1000;
         int cntOut = 10;
 
+        if (request.params.size() > 2 && request.params[2].isNum() && request.params[2].get_int() > 0)
```
Comment (tawmaz) [inline src/pocketdb/web/SearchRpc.cpp:348]: Same comment as function above, requrest.params.size(), return error if we get no parameters or too many.

File: src/pocketdb/consensus/social/BoostContent.hpp [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+    /*******************************************************************************************************************
+    *  BoostContent consensus base class
+    *******************************************************************************************************************/
+    class BoostContentConsensus : public SocialConsensus<BoostContent>
+    {
+    public:
+        BoostContentConsensus(int height) : SocialConsensus<BoostContent>(height) {}
+
+        ConsensusValidateResult Validate(const CTransactionRef& tx, const BoostContentRef& ptx, const PocketBlockRef& block) override
```
Comment (tawmaz) [inline src/pocketdb/consensus/social/BoostContent.hpp:24]: Since we are adding these from new I would like to see unit tests added which verify correct boost transaction and invalid boost transaction with edge cases tested.

**Review by tawmaz — APPROVED** [review]
Would like to see unit tests added for post boost.

File: src/pocketdb/web/SearchRpc.cpp (hunk 1 above)
Comment (andyoknen) [reply src/pocketdb/web/SearchRpc.cpp:288]: In fact, no - overload [] detects the presence of a field and returns NullableUniValue in its absence

File: src/pocketdb/consensus/social/BoostContent.hpp (hunk 3 above)
Comment (andyoknen) [reply src/pocketdb/consensus/social/BoostContent.hpp:24]: Could you add test cases for this new transaction?

File: src/pocketdb/web/SearchRpc.cpp (hunk 1 above)
Comment (andyoknen) [reply src/pocketdb/web/SearchRpc.cpp:288]: https://github.com/pocketnetteam/pocketnet.core/blob/a2475b0bdd203521b5c7f888bea00b3092d16d8c/src/univalue/lib/univalue.cpp#L197

**Comment by andyoknen** [comment]
We need to add unit tests for a new transaction

**Comment by andyoknen** [comment]
@only1question There is not enough output logic in the feed for this transaction. We also did not discuss the limits and the minimum amount.
